app/src/main/kotlin/com/mapzen/erasermap/view/InstructionAdapter.kt

```diff
@@ -0,0 +1,95 @@
+package com.mapzen.erasermap.view
+
+import android.content.Context
+import android.support.v4.view.PagerAdapter
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.TextView
+import com.mapzen.erasermap.PrivateMapsApplication
+import com.mapzen.erasermap.R
+import com.mapzen.erasermap.model.RoutePreviewEvent
+import com.mapzen.erasermap.util.DisplayHelper
+import com.mapzen.pelias.SimpleFeature
+import com.mapzen.pelias.gson.Feature
+import com.mapzen.valhalla.Instruction
+import com.squareup.otto.Bus
+import javax.inject.Inject
+import java.util.ArrayList
+
+public class InstructionAdapter(val context: Context, val instructions: ArrayList<Instruction>,
+                                val pager: RouteModeView) : PagerAdapter() {
+
+    var instructionPager: RouteModeView = pager
+
+    override fun instantiateItem(container: ViewGroup?, position: Int): Any? {
+        val instruction = instructions.get(position)
+        val view = View.inflate(context, R.layout.pager_item_instruction, null)
+        val title = view.findViewById(R.id.instruction_text) as TextView
+        val distance = view.findViewById(R.id.distance) as TextView
+        val icon = view.findViewById(R.id.icon) as ImageView
+        var iconId: Int = DisplayHelper.getRouteDrawable(context,
+                instruction.getIntegerInstruction())
+        distance.setText((instruction.getFormattedDistance()))
+        title.setText(instruction.getHumanTurnInstruction())
+        icon.setImageResource(iconId)
+        if (position == 0) {
+            view.findViewById(R.id.left_arrow).setVisibility(View.INVISIBLE)
+        }
+        if (position == getCount() - 1) {
+            view.findViewById(R.id.right_arrow).setVisibility(View.INVISIBLE)
+        }
+        initArrowOnClickListeners(view, position)
+        setTagId(view, position)
+        container?.addView(view)
+        return view
+    }
+
+    fun getView(): View {
+        return this.getView()
+    }
+
+    override fun getCount(): Int {
+        return instructions.size()
+    }
+
+    private fun setTagId(view: View, position: Int) {
+        view.setTag("Instruction_" + position)
+    }
+
+    private fun initArrowOnClickListeners(view: View, position: Int) {
+        view.findViewById(R.id.right_arrow).setOnClickListener({
+            instructionPager.pageForward(position)
+        })
+        view.findViewById(R.id.left_arrow).setOnClickListener({
+            instructionPager.pageBackwards(position)
+        })
+    }
+
+    public fun setBackgroundColorActive(view: View?) {
+        if (view != null) {
```

Comment:
If statement can be replaced with safe access operator here. Same result just makes it a little more concise.

        view?.setBackgroundColor(context.getResources().getColor(R.color.transparent_white))